experimental/dds/tree/src/Checkout.ts

```diff
@@ -172,39 +172,51 @@ export abstract class Checkout extends EventEmitterWithErrorHandling<ICheckoutEv
 		assert(currentEdit !== undefined, 'An edit is not open.');
 		this.currentEdit = undefined;
 		const editingResult = currentEdit.close();
```

Comment:
We may want to check the status of the `currentEdit` before calling `.close()`:
- If if the status is good then we proceeed with the new code. (Which may log a full error if `.close()` fails)
- If the status is bad then we could just throw a "Can't close a failed transaction" error.

In the "status is bad" case, your changes ensure that the failure telemetry would have been logged already.